File: cpp/program43.cpp
// program24.c

#include<iostream>
using namespace std;

class Number
{
    public:
        int iNo;

    Number(int X)
    {
        iNo = X;
    }

    void DisplayFactors()
    {
        int iCnt = 0; 

        for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                cout<<iCnt<<"\n";
            }
        }
    }
};

int main()
{
    int iValue = 0;

    cout<<"Enter Number"<<"\n";
    cin>>iValue;

    Number nobj(iValue);
    nobj.DisplayFactors();

    return 0;
}

File: java/program43.java
import java.util.*;

class Number 
{
    public int iNo;

    public Number(int no)
    {
        iNo = no;
    }

    public void DisplayFactors()
    {
        for(int icnt = 1; icnt <= (iNo/2); icnt++)
        {
            if((iNo%icnt) == 0)
            {
                System.out.println(icnt);
            }
        }
    }
}

class program43
{
    public static void main(String arg[])
    {
        int iValue = 0;
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        iValue = sobj.nextInt();

        Number obj = new Number(iValue);
        obj.DisplayFactors();
    }
}